lib/Transforms/Scalar/PromoteVerifierCalls.cc

```diff
@@ -209,6 +213,33 @@ bool PromoteVerifierCalls::runOnFunction(Function &F) {
       if (cg)
         (*cg)[&F]->addCalledFunction(ci, (*cg)[ci->getCalledFunction()]);
 
+      I.replaceAllUsesWith(ci);
+      toKill.push_back(&I);
+    } else if (fn && (fn->getName().equals("sea_is_read"))) {
+      IRBuilder<> Builder(F.getContext());
```

Comment:
the bodies of these if statements are all the same, why not refactor the code into a (lambda) function and set a map between function name and function that replaces it. So the code looks like that:
```
if (map.count(fn->getName())) {
   replace(fn, map.at(fn->getName());
 }
```